src/modules/LimitOrders.ts

```diff
@@ -0,0 +1,353 @@
+  'uint256 makerAccountNumber,' +
+  'address takerAccountOwner,' +
+  'uint256 takerAccountNumber,' +
+  'uint256 expiration,' +
+  'uint256 salt' +
+  ')';
+
+export class LimitOrders {
+  private contracts: Contracts;
+  private web3: Web3;
+  private networkId: number;
+
+  // ============ Constructor ============
+
+  constructor(
+    contracts: Contracts,
+    web3: Web3,
+    networkId: number,
+  ) {
+    this.contracts = contracts;
+    this.web3 = web3;
+    this.networkId = networkId;
+  }
+
+  // ============ Sender Contract Methods ============
+
+  public async approveOrder(
+    order: LimitOrder,
+    options?: ContractCallOptions,
+  ): Promise<any> {
+    const orderHash = this.getOrderHash(order);
+    return await this.contracts.callContractFunction(
+      this.contracts.limitOrders.methods.approveOrder(orderHash),
+      options,
+    );
+  }
+
+  public async cancelOrder(
+    order: LimitOrder,
+    options?: ContractCallOptions,
+  ): Promise<any> {
+    const orderHash = this.getOrderHash(order);
+    return await this.contracts.callContractFunction(
+      this.contracts.limitOrders.methods.cancelOrder(orderHash),
+      options,
+    );
+  }
+
+  // ============ Getter Contract Methods ============
+
+  public async getOrderStatus(
+    order: LimitOrder,
+    options?: ContractConstantCallOptions,
+  ): Promise<{status: number, makerFilledAmount: Integer}> {
+    const orderHash = this.getOrderHash(order);
+
+    const [
+      status,
+      makerFilledAmount,
+    ] = await Promise.all([
+      this.contracts.callConstantContractFunction(
+        this.contracts.limitOrders.methods.g_status(
+          order.makerAccountOwner,
+          orderHash,
+        ),
+        options,
+      ),
+      this.contracts.callConstantContractFunction(
+        this.contracts.limitOrders.methods.g_filled(
+          orderHash,
+        ),
+        options,
+      ),
+    ]);
+
+    return {
+      status: parseInt(status, 10),
+      makerFilledAmount: new BigNumber(makerFilledAmount),
+    };
+  }
+
+  // ============ Signing Methods ============
+
+  public async ethSignTypedOrderV3(
+    order: LimitOrder,
+    hashOptions: any = {},
+  ): Promise<string> {
+    return this.ethSignTypedOrderInternal(
+      order,
+      'eth_signTypedData_v3',
+      hashOptions,
+    );
+  }
+
+  public async ethSignTypedOrder(
+    order: LimitOrder,
+    hashOptions: any = {},
+  ): Promise<string> {
+    return this.ethSignTypedOrderInternal(
+      order,
+      'eth_signTypedData',
+      hashOptions,
+    );
+  }
+
+  public async ethSignOrder(
+    order: LimitOrder,
+    hashOptions: any = {},
+  ): Promise<string> {
+    const hash = this.getOrderHash(order, hashOptions);
+    const signature = await this.web3.eth.sign(hash, order.makerAccountOwner);
+    return createTypedSignature(signature, SIGNATURE_TYPES.NO_PREPEND);
+  }
+
+  public async ethSignCancelOrder(
+    order: LimitOrder,
+    hashOptions: any = {},
```

Comment:
can you make this not have `any` type? otherwise it's impossible to tell what you're supposed to pass in

Reply:
got rid of hashOptions